controllers/external-accounts.js

```diff
@@ -71,14 +71,20 @@ const linkExternalAccount = async (req, res) => {
       userId
     );
 
-    try {
-      const unverifiedRoleId = config.get("discordUnverifiedRoleId");
-      await discordServices.removeRoleFromUser(unverifiedRoleId, attributes.discordId, req.userData);
-    } catch (error) {
-      logger.error(`Error getting external account data: ${error}`);
-      return res.boom.internal("Role Deletion failed. Please contact admin.", {
-        message: "Role Deletion failed. Please contact admin.",
-      });
+    const unverifiedRoleId = config.get("discordUnverifiedRoleId");
+    const unverifiedRoleRemovalResponse = await removeDiscordRoleUtils.removeDiscordRoleFromUser(
+      req.userData,
+      attributes.discordId,
+      unverifiedRoleId
+    );
+
+    if (!unverifiedRoleRemovalResponse.success) {
+      return res.boom.internal(
+        `User details updated but ${unverifiedRoleRemovalResponse.message}. Please contact admin`,
```

Comment:
Why this is not in try catch block?

Reply:
Because within the function I've already specified the return statements. So it will always get a success or failure with a message.